tests/test_signal_generator.cpp

```diff
@@ -0,0 +1,216 @@
+#include <hyro/SignalGenerator.h>
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/Thresholding.h>
+#include <hyro/DigitalConverterComponent.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+
+#include <hyro/test/ComponentTester.h>
+#include <gtest/gtest.h>
+
+namespace hyro
+{
+namespace test
+{
+
+//-------------------------------------------------
+// Create test for Dynamic Properties
+//-------------------------------------------------
+TEST (ExampleTest, DynamicPropertiesCheck)
+{
+  //Configuration
+  auto signal_conf = "{"
+    "outputs: {"
+      "value: { protocol: 'api' },"
+      "timestamp: { protocol: 'api' },"
+      "frame_id: { protocol: 'api' },"
+      "fix_dynamic: { protocol: 'api' }"
+      "}"
+    "}";
+
+  auto digital_conf = "{"
+    "outputs: {"
+      "digital: { protocol: 'api' },"
+      "fix_dynamic: { protocol: 'api' }"
+    "},"
+    "inputs: {"
+      "value: { protocol: 'api' }"
+  "}}";
+
+  //SignalGeneratorComponent and DigitalConverterComponent state machines
+  StateMachine sm(std::make_shared<SignalGeneratorComponent>("/signal"_uri));
```

Comment:
Please, give a meaningful name for the variables. e.g.: `signal_generator_sm`, `digital_converter_sm`.